packages/eas-cli/src/build/android/version.ts

```diff
@@ -2,13 +2,92 @@ import { ExpoConfig } from '@expo/config';
 import { AndroidConfig } from '@expo/config-plugins';
 import { Platform, Workflow } from '@expo/eas-build-job';
 import { BuildProfile } from '@expo/eas-json';
+import chalk from 'chalk';
+import fs from 'fs-extra';
 
+import Log from '../../log';
 import {
   getAppBuildGradleAsync,
   parseGradleCommand,
   resolveConfigValue,
 } from '../../project/android/gradleUtils';
 import { resolveWorkflowAsync } from '../../project/workflow';
+import { updateAppJsonConfigAsync } from '../utils/appJson';
+import { bumpShortVersionAsync, ensureStaticConfigExists } from '../utils/version';
+
+export enum BumpStrategy {
+  SHORT_VERSION,
+  VERSION_CODE,
+  NOOP,
+}
+export async function bumpVersionAsync({
+  bumpStrategy,
+  projectDir,
+  exp,
+}: {
+  projectDir: string;
+  exp: ExpoConfig;
+  bumpStrategy: BumpStrategy;
+}): Promise<void> {
+  if (bumpStrategy === BumpStrategy.NOOP) {
+    return;
+  }
+  ensureStaticConfigExists(projectDir);
+
+  const buildGradle = await getAppBuildGradleAsync(projectDir);
+  const isMultiFlavor =
+    buildGradle.android?.productFlavors ||
+    (buildGradle.android?.flavorDimensions ?? '').split(',').length > 1;
+  if (isMultiFlavor) {
+    throw new Error('automatic version bumping is not supported for multiflavor android projects');
+  }
+
+  await bumpVersionInAppJsonAsync({ bumpStrategy, projectDir, exp });
+  Log.log('Updated versions in app.json');
+  await writeVersionsToBuildGradleAsync({
+    projectDir,
+    exp,
+  });
+  Log.log('Synchronized versions with build gradle');
+}
+
+export async function bumpVersionInAppJsonAsync({
+  bumpStrategy,
+  projectDir,
+  exp,
+}: {
+  bumpStrategy: BumpStrategy;
+  projectDir: string;
+  exp: ExpoConfig;
+}): Promise<void> {
+  ensureStaticConfigExists(projectDir);
+  Log.addNewLineIfNone();
+
+  switch (bumpStrategy) {
+    case BumpStrategy.NOOP: {
+      return;
+    }
+    case BumpStrategy.SHORT_VERSION: {
+      const shortVersion = AndroidConfig.Version.getVersionName(exp) ?? '1.0.0';
+      await bumpShortVersionAsync({ shortVersion, projectDir, exp });
+      break;
+    }
+    case BumpStrategy.VERSION_CODE:
+    default: {
+      const versionCode = AndroidConfig.Version.getVersionCode(exp);
```

Comment:
`getVersionCode` returns a `number` so we don't need to do the semver parsing done for the iOS `buildNumber`.